pkg/network/vmispec/interface.go

```diff
@@ -120,11 +120,10 @@ func IndexInterfaceSpecByMac(interfaces []v1.Interface) map[string]v1.Interface
 	return ifacesByMac
 }
 
-func LookupInterfaceByNetwork(ifaces []v1.Interface, network *v1.Network) *v1.Interface {
-	for _, iface := range ifaces {
-		if iface.Name == network.Name {
-			iface := iface
-			return &iface
+func LookupInterfaceByName(ifaces []v1.Interface, name string) *v1.Interface {
```

Comment:
That the caller will add/remove items from the list.

Reply:
The caller gets a pointer to the item, not to the list. How can she remove it?

Reply:
Since the item is part of a list. The returned pointer points to some internal structure that the list holds. Changes to the list, like adding and removing items can causes changes to the list internal structure like deaallocating the whole internal structure to another place in the memory. Which can lead the returned pointer to point to some unidentified place.